src/Operations.jl

```diff
@@ -14,12 +14,39 @@ import ..Artifacts: ensure_all_artifacts_installed, artifact_names, extract_all_
 using Base.BinaryPlatforms
 import ...Pkg
 import ...Pkg: pkg_server
+using Serialization
 
 #########
 # Utils #
 #########
 
+const PkgUUID = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
 const pkgs_precompile_suspended = Base.PkgId[]
+pkg_scratchpath() = joinpath(depots1(), "scratchspaces", PkgUUID)
+function save_suspended_packages()
+    path = pkg_scratchpath()
+    fpath = joinpath(path, string("suspend_cache_", hash(Base.active_project())))
+    try
```

Comment:
I think the try catch can be only around the `serialize`

Reply:
Yeah, thanks. I realised we don't need to try-catch the `serialize`, just the `deserialize`, and the file handling can be done via `isfile()`